src/provider/states/starting.rs

```diff
@@ -0,0 +1,120 @@
+use std::ffi::OsStr;
+use std::process::{Command, Stdio};
+
+use kubelet::pod::Pod;
+use kubelet::state::prelude::*;
+use kubelet::state::{State, Transition};
+use log::{debug, error, info, trace};
+use tokio::time::Duration;
+
+use crate::provider::states::create_config::CreatingConfig;
+use crate::provider::states::failed::Failed;
+use crate::provider::states::running::Running;
+use crate::provider::PodState;
+
+#[derive(Default, Debug, TransitionTo)]
+#[transition_to(Running, Failed)]
+pub struct Starting;
+
+#[async_trait::async_trait]
+impl State<PodState> for Starting {
```

Comment:
I did not really look at this file because I assume most of it will go away with systemd integration